xilem/src/view/board.rs

```diff
@@ -0,0 +1,384 @@
+    Widget,
+};
+use xilem_core::{
+    AnyElement, AnyView, AppendVec, DynMessage, ElementSplice, MessageResult, Mut, SuperElement,
+    View, ViewElement, ViewMarker, ViewSequence,
+};
+
+use crate::{Pod, ViewCtx, WidgetView};
+
+pub use masonry::widget::BoardParams;
+
+mod kurbo_shape;
+mod style_modifier;
+
+pub use kurbo_shape::GraphicsView;
+pub use style_modifier::{fill, stroke, transform, Fill, GraphicsExt, Stroke, Transform};
+
+pub fn board<State, Action, Seq: BoardSequence<State, Action>>(
+    sequence: Seq,
+) -> Board<Seq, State, Action> {
+    Board {
+        sequence,
+        phantom: PhantomData,
+    }
+}
+
+pub struct Board<Seq, State, Action = ()> {
+    sequence: Seq,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+impl<Seq, State, Action> ViewMarker for Board<Seq, State, Action> {}
+impl<State, Action, Seq> View<State, Action, ViewCtx> for Board<Seq, State, Action>
+where
+    State: 'static,
+    Action: 'static,
+    Seq: BoardSequence<State, Action>,
+{
+    type Element = Pod<widget::Board>;
+
+    type ViewState = Seq::SeqState;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let mut elements = AppendVec::default();
+        let mut widget = widget::Board::new();
+        let seq_state = self.sequence.seq_build(ctx, &mut elements);
+        for BoardElement { element } in elements.into_inner() {
+            widget = widget.with_child_pod(element.inner);
+        }
+        (Pod::new(widget), seq_state)
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        // TODO: Re-use scratch space?
+        let mut splice = BoardSplice::new(element);
+        self.sequence
+            .seq_rebuild(&prev.sequence, view_state, ctx, &mut splice);
+        debug_assert!(splice.scratch.is_empty());
+        splice.element
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<'_, Self::Element>,
+    ) {
+        let mut splice = BoardSplice::new(element);
+        self.sequence.seq_teardown(view_state, ctx, &mut splice);
+        debug_assert!(splice.scratch.into_inner().is_empty());
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[xilem_core::ViewId],
+        message: DynMessage,
+        app_state: &mut State,
+    ) -> MessageResult<Action> {
+        self.sequence
+            .seq_message(view_state, id_path, message, app_state)
+    }
+}
+
+pub type AnyBoardView<State, Action = ()> =
+    dyn AnyView<State, Action, ViewCtx, BoardElement> + Send + Sync;
+
+pub struct BoardElement {
+    element: Pod<Box<dyn SvgElement>>,
+}
+
+pub struct BoardElementMut<'w> {
+    parent: WidgetMut<'w, widget::Board>,
+    idx: usize,
+}
```

Comment:
And I think this could be removed as well, since I think we can just use the `ViewElement` implementation in the top of `xilem` of `Pod<W>`.

I believe we need something like: `impl<W: SvgElement> SuperElement<Pod<W>> for Pod<Box<dyn SvgElement>>`  then instead.

(It could well be that this needs some plumbing on the masonry side, in particular something like `WidgetPod<W: SvgElement>::svg_boxed()`)

The index here is already in the `BoardSplice`, the only other place I see where this is necessary, is for the `AnyElement` impl. But I think we can reuse the slightly modified `DynWidget`, roughly like this:

```rust
pub struct DynWidget<W = Box<dyn Widget>> {
    inner: WidgetPod<W>,
}
```

with `impl SvgElement for DynWidget<Box<dyn SvgElement>>`, which would make all of this more generally useful.

This would make a lot of code below unnecessary too.